handlers/main.yml

```diff
@@ -50,6 +50,14 @@
   command: auditctl -f {{ rhel7stig_auditd_failure_flag }}
   failed_when: no
 
+- name: clean up ssh host key
+  file:
+      path: "{{ item }}"
+      state: absent
+  with_items:
+      - /etc/ssh/ssh_host_rsa_key
+      - /etc/ssh/ssh_host_rsa_key.pub
+
```

Comment:
What are the implications of cleaning up this key? If I expect this role to install, enable, and configure OpenSSH Server then removing these keys means I will need to regenerate them again? Or will they get generated the next time ssh starts up?

Reply:
They are generated at boot time. I'm running the role at kickstart time, before they've been generated. I clean them up so that machines deployed in "gold image" style will still generate fresh keys at first boot. The role only cleans these keys if it created them.

Reply:
Right...some people might not be using the role in the same way and may expect that the keys exist (even if the role creates them). Typically in a kickstart setup, I would have all my cleanup items like this at the end, outside of this role. Maybe it should just be optional or something. `rhel7stig_cleanup_keys` or `rhel7stig_cleanup_for_gold` or something. The second option can be used if we have any other cleanup items that should be run before "gold".

Reply:
If we create the keys, we should clean them up, i'm not opinion. I don't feel comfortable that my invocation is fips compliant, and I'd rather let the system do it. I also create only one type of key, whereas the system generates 3.

I'd be okay gating this behind a flag, disabled by default, in which case the role fails on any ssh remediation.

Reply:
I thought OpenSSH would generate the host keys when it starts up if they don't exist. Can't we just start the service so it generates the keys? Then you can have a conditional clean up task and others that want to keep the keys can keep them?

Reply:
I'd be fine inserting 'service: name=sshd state=started' before the manual key creation, so that it will  auto-create the keys in any case except for running in a kickstart environment.  (starting services in the ks environment does not work, and we already have a flag to avoid trying. `system_is_chroot`)  The latest version of ansible detects this automatically and issue a warning if you try.

Reply:
I've changed my mind on this. On further consideration, it is very unlikely that someone is running this role against a system that does not have SSH installed unless they are doing it as part of kickstart or some other golden image creation process that will get a reboot after.